scipy/fft/tests/test_basic.py

```diff
@@ -114,11 +133,12 @@ def test_ifftn(self, xp):
 
     @array_api_compatible
     def test_rfft(self, xp):
-        x = xp.asarray(random(29))
+        x = xp.asarray(random(29), dtype=xp.float64)
         for n in [size(x), 2*size(x)]:
             for norm in [None, "backward", "ortho", "forward"]:
                 xp_assert_close(fft.rfft(x, n=n, norm=norm),
-                                fft.fft(x, n=n, norm=norm)[:(n//2 + 1)])
+                                fft.fft(xp.asarray(x, dtype=xp.complex128),
```

Comment:
```suggestion
                                fft.fft(xp.astype(x, dtype=xp.complex128),
```

Reply:
good point, we've used `astype` in other places already so should work fine and be more accurate